examples/jupyterhub/README.md

```diff
@@ -0,0 +1,131 @@
+- a local installation of JupyterHub, configured to launch notebooks on the Slurm cluster
+
+This deployment not only demonstrates the use case but also serves as a platform for testing and developing FirecRESTSpawner.
+
+
+## Requirements
+
+For this tutorial you will need
+
+ * a recent installation of Docker, which includes the `docker compose` command (or the older `docker-compose` command line tool)
+ * a Python installation (version 3.9 or higher)
+
+
+## Deployment of FirecREST and Slurm cluster
+
+This tutorial builds on the Docker demo of FirecREST.
+We will use the small [docker-compose-jhub.yaml](docker-compose-jhub.yaml) file to override some settings in the FirecREST demo.
+This can be done by passing both files to the `docker compose` command.
+
+First, we clone the FirecREST repository
+
+```bash
+git clone https://github.com/eth-cscs/firecrest-v2.git
+```
+
+and then launch the deployment
+
+```bash
+cd firecrest-v2
+export JHUB_DOCKERFILE_DIR=/path/to/tutorial
+docker compose -f docker-compose-minimal-env.yml -f /path/to/tutorial/docker-compose-jhub.yaml up
+```
+
+This step takes a few minutes. In the meanwhile we can install JupyterHub on a local virtual environment.
+
+
+### Install JupyterHub and FirecRESTSpawner
+
+An easy way to install JupyterHub is via [Miniconda](https://docs.anaconda.com/miniconda/install/).
+We need to [download the Miniconda installer](https://docs.anaconda.com/miniconda/install/) for our platforms and install it using the following command
+
+```bash
+bash Miniconda3-latest-<arch>.sh -p /path/to/mc-jhub -b
+```
+
+Here we use `-p` to pass the absolute path to the install directory and `-b` to accept the [terms of service](https://legal.anaconda.com/policies/en/).
+
+We can activate our conda base environment and install configurable-http-proxy, JupyterHub and FirecRESTSpawner
+
+```bash
+. /path/to/mc-jhub/bin/activate
+conda install -y configurable-http-proxy
+pip install --no-cache jupyterhub==4.1.6 pyfirecrest==2.6.0 SQLAlchemy==1.4.52 oauthenticator==16.3.1 python-hostlist==1.23.0
+git clone https://github.com/eth-cscs/firecrestspawner.git
+cd firecrestspawner
+git checkout apiv2
+pip install --no-cache .
+```
+
+## Back to the deployment
+
+Once the images have been built, you can check that all containers are running
+
+```bash
+docker compose -p firecrest-v2 ps --format 'table {{.ID}}\t{{.Name}}\t{{.State}}'
+```
+
+That should show something like this
+
+```bash
+CONTAINER ID   NAME                       STATE
+fd8b1575bf18   firecrest-v2-firecrest-1   running
+4a6e4c3d089d   firecrest-v2-keycloak-1    running
+36f8d98f3b67   firecrest-v2-minio-1       running
+85de3b4afe95   firecrest-v2-slurm-1       running
+```
+
+When we are done with the tutorial, the deployment can be shutdown by pressing `ctrl+c` and then
+
+```
+cd firecrest-v2
+docker compose -f docker-compose-minimal-env.yml -f /path/to/tutorial/docker-compose-jhub.yaml down
+```
+
+### Setting up the authorization
```

Comment:
You could simplify this step by including jhub client in the default keycloak setup: https://github.com/eth-cscs/firecrest-v2/blob/master/build/docker/keycloak/config.json